addons/crm_iap_lead/models/crm_iap_lead_mining_request.py

```diff
@@ -28,19 +29,29 @@ def _default_lead_type(self):
         else:
             return 'opportunity'
 
+    def _default_active_company(self):
+        return self.env.user.company_id.country_id
+
+    def _default_user(self):
+        return self.env.user
+
+    def _default_sales_team_current_user(self):
```

Comment:
Issue is that those default values could lead to incoherent data. If you want to have default value for user and team, you will have to make team_id an editable computed stored field, a bit like what is done in crm_lead. Basically when changing the user, we check if the sales team is coherent or we update it. By making it editable stored computed (readonly=False, compute='_compute_team_id', store=True), we let people change it in addition to an "onchange" behavior.